hypar-io/Elements
# Elements Support for the Hub

BACKGROUND:
In order to support the hub and visualization in context in Revit, updates are required to Elements.

DESCRIPTION:
- Various tessellate method signatures are updated to allow assigning colors at the time of tessellation as Revit requires vertex colors.
 - Updates are made to the type generator to support compiling multiple types into an assembly on disk.
- Hypar.Revit is completely rewritten as an external application, two external commands, and an `IDirectContext3D` server.

TESTING:
- Pull this branch.
- Build.
- Ensure that the `Hypar.Revit.addin` made it to your addins folder, or place it there.
- Build the CLI from this branch: https://github.com/hypar-io/Hypar/pull/211.
- Create a `~/.hypar/hub.json` file by looking at the example in the above pull request. Specify a workflow and an associated revit file.
- From the command line do `hypar hub`
- Run Revit. You should see two commands "Hypar Hub Start" and "Hypar Hub Stop" under External Commands.
- Once the hub is started, click the "Hypar Hub Start" command. On your next mouse move you should see the hub data visualized in Revit.

FUTURE WORK:
- This work only supports the visualization of data from the hub in Revit. "Baking" Revit elements will come in a future update.


---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/hypar-io/elements/340)
